src/RouteServiceAuth/Whitelist.cs

```diff
@@ -0,0 +1,84 @@
+namespace RouteServiceAuth
+{
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Options;
+
+    public interface IWhitelist
+    {
+        bool IsWhitelisted(HttpRequest request);
+    }
+
+    public sealed class Whitelist : IWhitelist
+    {
+        const string BaseAddress = "http://localhost";
+
+        private readonly ILogger _logger;
+        private readonly IOptionsMonitor<WhitelistOptions> _optionsMonitor;
+
+        public List<Uri> Entries { get;} = new List<Uri>();
+
+        public Whitelist(ILogger<Whitelist> logger, IOptionsMonitor<WhitelistOptions> optionsMonitor)
+        {
+            _logger = logger;
+            _optionsMonitor = optionsMonitor;
+            _optionsMonitor.OnChange(options => Bind(options));
+            Bind(_optionsMonitor.CurrentValue);
+        }
+
+        void Bind(WhitelistOptions options)
+        {
+            Entries.Clear();
```

Comment:
Direct manipulation of Entries is dangerous as it's not a thread safe collection. Should build up a new collection in variable and then replace Entries with new value, which is atomic and would not require locks.

Reply:
Good catch - implemented as recommended.